test/DynamicHelpTest.cs

```diff
@@ -0,0 +1,168 @@
+using System;
+using System.Collections;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Management.Automation;
+using System.Reflection;
+using System.Text;
+using Microsoft.PowerShell;
+using Xunit;
+
+namespace Test
+{
+    public partial class ReadLine
+    {
+        private static string actualContent;
+
+        private static readonly string fullHelp = @"
+
+NAME
+    Get-Date
+
+SYNOPSIS
+    Gets the current date and time.
+
+PARAMETERS
+    -AsUTC <System.Management.Automation.SwitchParameter>
+        Converts the date value to the equivalent time in UTC.
+
+        This parameter was introduced in PowerShell 7.1.
+
+        Required?                    false
+        Position?                    named
+        Default value                False
+        Accept pipeline input?       False
+        Accept wildcard characters?  false
+";
+
+        internal static object GetDynamicHelpContent(string commandName, string parameterName, bool isFullHelp)
+        {
+            string descText = @"Specifies a date and time. Time is optional and if not specified, returns 00:00:00.";
+
+            if (!String.IsNullOrEmpty(commandName) && isFullHelp)
+            {
+                return fullHelp;
+            }
+            else if (!String.IsNullOrEmpty(commandName) && !String.IsNullOrEmpty(parameterName))
```

Comment:
Please use `string` instead of `String` in new code, so as to be consistent with rest of the code base.
```suggestion
            else if (!string.IsNullOrEmpty(commandName) && !String.IsNullOrEmpty(parameterName))
```

Reply:
fixed